### PR TITLE
Changes to survey-runner to use sdc-cryptography library

### PR DESCRIPTION
**Pull request by shahi645**
### What is the context of this PR?
- removal of current eq cryptography code
- simplify code where possible
- update tests

### How to review 
Look through changes to code and run tests

### CODE REVIEW

File: tests/integration/views/test_jwt_authentication.py [hunk 1]
```diff
@@ -28,6 +30,8 @@ def test_no_token(self):
 
     def test_invalid_token(self):
         token = "invalid"
+        # token = token.encode('utf-8')
```
Comment (jcox-dev) [inline tests/integration/views/test_jwt_authentication.py]: Extra comment from dev left in.

File: app/authentication/authenticator.py [hunk 2]
```diff
@@ -81,14 +82,14 @@ def decrypt_token(encrypted_token):
...
+                                  purpose=KEY_PURPOSE_AUTHENTICATION)
 
-    decrypted_token = decode_jwt(jwt_token=jwt_token,
-                                 secret_store=current_app.eq['secret_store'],
-                                 purpose=KEY_PURPOSE_AUTHENTICATION,
-                                 leeway=current_app.config['EQ_JWT_LEEWAY_IN_SECONDS'])
+    decrypted_token = JWTHelper.decode(jwt_token=jwt_token,
+                                       key_store=current_app.eq['secret_store'],
+                                       purpose=KEY_PURPOSE_AUTHENTICATION,
+                                       leeway=current_app.config['EQ_JWT_LEEWAY_IN_SECONDS'])
```
Comment (jcox-dev) [inline app/authentication/authenticator.py]: This repeated pattern of `decrypt` -> `decode` can be dropped in favour of the helper function from the lib, `sdc.crypto.decrypter.decrypt`.

File: tests/integration/create_token.py [hunk 3]
```diff
@@ -63,6 +65,6 @@ def create_token_with_survey_url(self, form_type_id, eq_id, survey_url, **extra_
         return self.generate_token(payload_vars)
 
     def generate_token(self, payload):
-        token = encode_jwt(payload, self._upstream_kid, self._secret_store, KEY_PURPOSE_AUTHENTICATION)
+        token = JWTHelper.encode(payload, self._upstream_kid, self._secret_store, KEY_PURPOSE_AUTHENTICATION)
 
-        return encrypt_jwe(token, self._sr_public_kid, self._secret_store, KEY_PURPOSE_AUTHENTICATION)
+        return JWEHelper.encrypt(token, self._sr_public_kid, self._secret_store, KEY_PURPOSE_AUTHENTICATION)
```
Comment (jcox-dev) [inline tests/integration/create_token.py]: Likewise, `encode` -> `encrypt` can be dropped in favour of `sdc.crypto.encrypter.encrypt`.

**Review by jcox-dev — CHANGES_REQUESTED** [review]
Change to use the helper methods `encrypt` and `decrypt` from the sdc-crypto lib and clean up extra comment. 

Attached suggested patch: [helper.txt](https://github.com/ONSdigital/eq-survey-runner/files/1333888/helper.txt).

File: tests/integration/create_token.py (hunk 3 above)
Comment (jcox-dev) [reply tests/integration/create_token.py]: Attached suggested patch [here](https://github.com/ONSdigital/eq-survey-runner/files/1333883/helper.txt).

**Review by jcox-dev — APPROVED** [review]

File: app/views/session.py [hunk 4]
```diff
@@ -50,6 +51,8 @@ def login():
         jti_claim_storage.use_jti_claim(jti_claim)
     except JtiTokenUsed as e:
         raise Unauthorized from e
+    except (TypeError, ValueError) as e:
+        raise InvalidTokenException
```
Comment (ajmaddaford) [inline app/views/session.py]: This should be `raise InvalidTokenException from e` to keep the traceback from the original exception.

**Review by ajmaddaford — CHANGES_REQUESTED** [review]

File: app/secrets.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
 from jwcrypto import jwk
```
Comment (jonnyshaw89) [inline app/secrets.py:1]: All of this logic has also been replaced in the new library. or are we blocked from doing this until the secrets and keys have been split?

File: tests/app/authentication/test_invalid_token_exception.py [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import unittest
```
Comment (jonnyshaw89) [inline tests/app/authentication/test_invalid_token_exception.py]: This test now seems pointless?we are testing an exception that is provided by the library.

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/secrets.py (hunk 5 above)
Comment (shahi645) [reply app/secrets.py:1]: yeah this is blocked until we split the keys from secrets. Card in ready for dev currently

File: tests/app/authentication/test_invalid_token_exception.py (hunk 6 above)
Comment (shahi645) [reply tests/app/authentication/test_invalid_token_exception.py]: removed

**Review by jonnyshaw89 — APPROVED** [review]

**Review by ajmaddaford — APPROVED** [review]

**Pull request merged**